jax/experimental/sparse/bcoo.py

```diff
@@ -572,6 +576,52 @@ def _bcoo_dot_general_abstract_eval(lhs_data, lhs_indices, rhs, *, dimension_num
   out_dtype = jnp.promote_types(lhs_data.dtype, rhs.dtype)
   return core.ShapedArray(out_shape, out_dtype)
 
+
+_bcoo_dot_general_gpu_translation_rule_default = xla.lower_fun(
+    _bcoo_dot_general_impl, multiple_results=False, new_style=True)
+
+def _bcoo_dot_general_gpu_translation_rule(
+    ctx, avals_in, avals_out, lhs_data, lhs_indices, rhs, *, dimension_numbers,
+    lhs_shape):
+
+  (lhs_contract, rhs_contract), (lhs_batch, rhs_batch) = dimension_numbers
+  _, lhs_indices_aval, rhs_aval, = avals_in
+  n_batch, n_sparse, n_dense, _ = _validate_bcoo_indices(lhs_indices_aval, lhs_shape)
+
+  if n_batch or n_dense or n_sparse != 2 or rhs_aval.ndim > 2 or lhs_batch or rhs_batch or not lhs_contract:
+    return _bcoo_dot_general_gpu_translation_rule_default(
+      ctx, avals_in, avals_out, lhs_data, lhs_indices, rhs,
+      dimension_numbers=dimension_numbers, lhs_shape=lhs_shape)
```

Comment:
Design-wise, it occurred to me that it might be cleaner to put the cusparse logic in another function, so then this translation rule would just contain something like
```
if <conditions>:
  return _bcoo_dot_general_cuda_translation_rule(...)
else:
  return _bcoo_dot_general_default_translation_rule(...)
```
What do you think?